### ISSUE TITLE
owncloud-server-8.0.0-22.1 update issue

### ISSUE THREAD

**Issue by MaDYeTTi** [owncloud/core#14362]
Error while updating to owncloud-server-8.0.0-22.1 

  Updating   : owncloud-server-8.0.0-22.1.noarch                          11/76
owncloud-server: Upgrade starting
owncloud-server}: occ maintenance:mode --on
Maintenance mode enabled
owncloud-server Upgrade complete
su: unrecognized option '--off'
Try `su --help' for more information.

seems like some error in update script.

CentOS 6.5 + Apache

**Comment by chihoko** [comment]
Same issue on Centos 6.6 + Apache.
Results in site left in maintenance mode until config.php is manually updated.
Apache process has write permission to config.php.

**Comment by rbu** [comment]
Same on CentOS 7. Full log:

```
owncloud-server: Upgrade starting
owncloud-server}: occ maintenance:mode --on
Maintenance mode enabled
owncloud-server Upgrade complete
su: unrecognized option '--off'

Usage:
 su [options] [-] [USER [arg]...]

Change the effective user id and group id to that of USER.
A mere - implies -l.   If USER not given, assume root.

Options:
 -m, -p, --preserve-environment  do not reset environment variables
 -g, --group <group>             specify the primary group
 -G, --supp-group <group>        specify a supplemental group

 -, -l, --login                  make the shell a login shell
 -c, --command <command>         pass a single command to the shell with -c
 --session-command <command>     pass a single command to the shell with -c
                                 and do not create a new session
 -f, --fast                      pass -f to the shell (for csh or tcsh)
 -s, --shell <shell>             run shell if /etc/shells allows it

 -h, --help     display this help and exit
 -V, --version  output version information and exit

For more details see su(1).
owncloud update: Checking for running Apache
Redirecting to /bin/systemctl status  httpd.service
Redirecting to /bin/systemctl stop  httpd.service
owncloud pre-install: Stopping Apache
owncloud-config-apache: Upgrade complete
owncloud-config-apache: Restarting
Redirecting to /bin/systemctl start  httpd.service
An ownCloud installation consists of multiple packages.
You just uninstalled owncloud . This is what remains installed:
owncloud-3rdparty-8.0.0-20.1.noarch
owncloud-app-external-8.0.0-20.1.noarch
owncloud-app-user_webdavauth-8.0.0-20.1.noarch
owncloud-app-files_versions-8.0.0-20.1.noarch
owncloud-app-gallery-8.0.0-20.1.noarch
owncloud-app-files_sharing-8.0.0-20.1.noarch
owncloud-app-files-8.0.0-20.1.noarch
owncloud-3rdparty-8.0.0-22.1.noarch
owncloud-app-gallery-8.0.0-22.1.noarch
owncloud-app-templateeditor-8.0.0-22.1.noarch
owncloud-app-files_external-8.0.0-22.1.noarch
owncloud-app-files_locking-8.0.0-22.1.noarch
owncloud-app-updater-8.0.0-22.1.noarch
owncloud-server-8.0.0-20.1.noarch
owncloud-app-templateeditor-8.0.0-20.1.noarch
owncloud-app-files_external-8.0.0-20.1.noarch
owncloud-app-files_encryption-8.0.0-20.1.noarch
owncloud-config-apache-8.0.0-20.1.noarch
owncloud-app-firstrunwizard-8.0.0-20.1.noarch
owncloud-app-user_external-8.0.0-22.1.noarch
owncloud-app-firstrunwizard-8.0.0-22.1.noarch
owncloud-app-files_pdfviewer-8.0.0-22.1.noarch
owncloud-app-files_encryption-8.0.0-22.1.noarch
owncloud-app-user_ldap-8.0.0-22.1.noarch
owncloud-app-user_ldap-8.0.0-20.1.noarch
owncloud-app-files_locking-8.0.0-20.1.noarch
owncloud-app-files_trashbin-8.0.0-20.1.noarch
owncloud-app-updater-8.0.0-20.1.noarch
owncloud-app-files_videoviewer-8.0.0-20.1.noarch
owncloud-app-external-8.0.0-22.1.noarch
owncloud-app-provisioning_api-8.0.0-22.1.noarch
owncloud-app-files-8.0.0-22.1.noarch
owncloud-app-files_trashbin-8.0.0-22.1.noarch
owncloud-app-activity-8.0.0-22.1.noarch
owncloud-app-files_sharing-8.0.0-22.1.noarch
owncloud-app-user_external-8.0.0-20.1.noarch
owncloud-app-files_texteditor-8.0.0-20.1.noarch
owncloud-app-activity-8.0.0-20.1.noarch
owncloud-app-files_pdfviewer-8.0.0-20.1.noarch
owncloud-app-provisioning_api-8.0.0-20.1.noarch
owncloud-server-8.0.0-22.1.noarch
owncloud-config-apache-8.0.0-22.1.noarch
owncloud-app-files_texteditor-8.0.0-22.1.noarch
owncloud-app-files_videoviewer-8.0.0-22.1.noarch
owncloud-app-files_versions-8.0.0-22.1.noarch
owncloud-app-user_webdavauth-8.0.0-22.1.noarch
```

**Comment by LukasReschke** [comment]
@jnweiger

**Comment by gexat** [comment]
same issue with the last 3 updates on CentOS 6 & 7

**Comment by awaw** [comment]
--- owncloud.spec.orig  2015-02-23 10:28:01.524882239 +0100
+++ owncloud.spec       2015-02-23 10:31:01.331047598 +0100
@@ -677,10 +677,10 @@
 if [ -s /tmp/occ_maintenance_mode_during_owncloud_install ]; then
   if [ -x /usr/bin/php -a -f %{oc_dir}/occ ]; then
     # https://github.com/owncloud/core/issues/14351
-    su %{oc_user} -s /bin/sh -c "%{oc_dir}/occ maintenance:mode --off
+    su %{oc_user} -s /bin/sh -c "%{oc_dir}/occ maintenance:mode --off"
     echo "%{name}-server}: occ upgrade"
     su %{oc_user} -s /bin/sh -c "%{oc_dir}/occ upgrade"
-    su %{oc_user} -s /bin/sh -c "%{oc_dir}/occ maintenance:mode --off
+    su %{oc_user} -s /bin/sh -c "%{oc_dir}/occ maintenance:mode --off"
   fi
 fi
 rm -f /tmp/occ_maintenance_mode_during_owncloud_install

**Comment by DeepDiver1975** [comment]
fixed in https://github.com/owncloud/core/issues/14511 - this is actually a duplicate of https://github.com/owncloud/core/issues/14511 -> closing - THX

refs https://build.opensuse.org/package/rdiff/isv:ownCloud:community:testing/owncloud?linkrev=base&rev=60

**Comment by jnweiger** [comment]
fixes pushed to isv:ownCloud:community:8.0 -- happy testing!
